### ISSUE TITLE
Faster to answer bike lane quest (particularly "no bike lane")

### ISSUE THREAD

**Issue by smichel17** [westnordost/StreetComplete#2193]
**Use case**
Particularly in the rural US, roads outside of town are unlikely to have a bike lane. Yet, this still requires 5 taps (left, none, right, none, confirm), each of which introduces a context change (popup opening/closing) — a mildly annoying (enough to post this, anyway) amount of work to tag that a feature does *not* exist, which would be the default assumption anyway. …though admittedly, this is not the highest impact improvement because the quest is asked relatively few times (compared to [building heights](https://github.com/westnordost/StreetComplete/issues/1772), for example).

**Proposed Solution**

Options I can think of, and their problems:

- Add a box to pre-fill the last selected answer.
    - …how to decide which side of the road to fill with which answer? Maybe this is only useful for the "no answer option".
- Make the box always pre-fill "none", or just a button to one-click submit "none". Or, like the building height quest, allow submitting before both sides have been filled out and assume no lane by default.
    - Makes it easier to fill out one answer than the others; in some places, having a bike lane is more common than not.
- Update the interface to use fewer modals. It would be much faster to answer the quest if I could just tap instead of tap-wait-tap-wait…
    - Probably more effort than it is worth, to optimize this quest, until other higher-reward issues are fixed.

**Comment by FloEdelmann** [comment]
I actually had the same thought regarding the sidewalks quest today. There, the following would reduce the number of taps:

In the main form, have three options:

1. No sidewalks (1 tap instead of 4)
2. Sidewalks on both sides (1 tap instead of 4)
3. Sidewalk only on one side → opens modal to select which side, similar to the "which direction is up on these stairs" quest modal (2 taps instead of 4)

Unfortunately, this doesn't work for the bike lane quest, because you also need to specify the type of the bike lane. But maybe the idea can be adapted?

**Comment by smichel17** [comment]
So, I think the biggest culprit is actually the modals. I left out my specific idea for a UI change because I wasn't sure it was worth the effort, but given there's more than one quest it applies to…

For the sidewalk quest, you could instead show the same UI on start, but make tapping on a side toggle it directly; no menu at all. So, on each side, for tap number:

1. :grey_question: ⇒ :x:
2. :x: ⇒ :heavy_check_mark:
3. :heavy_check_mark: ⇒ :x:
…etc

This makes 1 tap per side with no sidewalk, and 2 taps per side with a sidewalk (but really, one double-tap, and barely any distance to move your finger — no need to stretch or re-adjust your grip). Could reduce by 1 tap each if you wanted to pre-fill both with :x: instead of :grey_question: (but I don't think the *number* of taps is the most important thing).

I think the equivalent UI paradigm for bike lanes is something like the fill tool ![image](https://user-images.githubusercontent.com/6563664/97064580-765cc100-1596-11eb-86a3-fc20efee23a3.png). Select the surface you want to "draw", then fill in each side that has it. However, this is a new paradigm for StreetComplete, with all the baggage (UX and implementation) that carries.

**Comment by westnordost** [comment]
I do not at all see the necessity to add any shortcurts to specifying the left side and the right side for cycleways and sidewalks. It may even have a negative effect to do that. The questions specifically ask about the left side and about the right side separately and it is tagged also this way (that the StreetComplete user checked for the left side and the right side individually). So, data users trust on that things tagged with `cycleway:left=X` and `cycleway:right=X` really were checked for both sides (unlike `cycleway=X`).

Keep in mind, that unlike for buildings (for housenumber quest etc ...) which you should be able to pass quite quickly while walking, the question for sidewalks and cycleways should be checked and answered for the whole street. So, the fact that it takes longer to answer this quest has much less a potential to impede the surveyor in his walking because the time to ascertain this data is usually much longer anyway than the time to input the data. The quest is just asked 1 time (up to a few times) per street, whilst for example the housenumber quest is asked for pretty much every building in the street, this is a big difference.

For some roads, the user actually has to cross the road to ascertain the cycleway situation, for example if there is a parking lane that blocks the view onto a possible cycle track. If the app provided a shortcut to tag "it is the same left and right" or "there is nothing here", it would facilitate a sloppy survey.

First of all, @smichel17 , the #1 UX principle of StreetComplete is that the UI is simple and self-explanatory, not that it is efficient. What you propose is not self-explanatory and if it is prefilled with ❌, there is even the danger that people will just tap "OK" without checking as pointed out in the last paragraph (or accidentally).

**Comment by westnordost** [comment]
Also, as a bonus, after you once specified the cycleway situation, and be it "no" everywhere, when the app asks again (in 4-8 years), you'll be able to simply select "yes, it's still the same". That's one tap then.

**Comment by matkoniecz** [comment]
> So, the fact that it takes longer to answer this quest has much less a potential to impede the surveyor in his walking because the time to ascertain this data is usually much longer anyway than the time to input the data.

It is less true in cases where there is cycling infrastructure anywhere and roads are quite heavily segmented (typical in my mapping, making this quest a bit irritating).

I even implemented in my fork some hacky solution to make answering `cycleway:both=no` quicker.

**Comment by smichel17** [comment]
@westnordost I agree with the UI criticisms; they are why I did not propose that particular solution when I opened the issue. I was/am mostly interested in a "quick fix" to enter "no bike lanes on either side" in areas where few roads have bike lanes. But given the drawbacks, and since I have no brilliant UI ideas to avoid them, I will close this issue for now — it is probably not worth spending more time to fix it until StreetComplete is basically complete and we are just polishing minor details.
